### ISSUE TITLE
shellcraft.mov does not support AH, BH, etc. registers

### ISSUE THREAD

**Issue by david942j** [Gallopsled/pwntools#1046]
```python
from pwn import *
# context.arch = 'i386'
print shellcraft.mov('eax', 'ah')
```
Expected:
`movzx eax, ah`

Result:
```
NameError: name 'ah' is not defined
```

The error is raised when trying to eval `ah` as a system-constant value

But the real reason is `ah` is not defined in `shellcraft/registers.py` as a register in `i386`.

Issue occurs in `amd64` as well.

**Comment by zachriggle** [comment]
I don't think we support the `xh` registers since they complicate some of the logic.  If you can submit a pull request that adds the functionality, and some doctests for various permutations (e.g. `mov("ah", "al")`) we can get this included.
